client/src/context/reducer.js

```diff
@@ -1,9 +1,11 @@
 export const initialState = {
   authenticated: false,
   user: {},
+  cart: { flights: [], hotels: [], rentalCar: [] },
 };
 
 export const userReducer = (initialState, action) => {
```

Comment:
Be careful here. I would rename this `initialState` as `state`. Because otherwise it could be confusing, you're not working with the `initialState` that's on line 1, although you could initialize it as a default:

```javascript
const userReducer = (state = initialState, action) => { ... }
```